From e71a208aab9f1530be80f6405bd0596dfca35a3a Mon Sep 17 00:00:00 2001
From: Alad Wenter <alad@archlinux.org>
Date: Wed, 7 Aug 2019 00:47:21 +0200
Subject: [PATCH] Also pass screen brightness keys as media keys

Rebase of upstream commits:

 2a6545e5bbc2d5d1f20ac71a45cabef27113d7cd # Also pass screen brightness keys as media keys 
 2766a1380818d70b7b8ed41b6aaca1124e11c790 # Add docs about passing of media keys 
 71cf4bda1b6f37847fae95de2f035995454743cd # Add long option for media key passing to docs 
---
 i3lock.1 | 6 ++++++
 i3lock.c | 2 ++
 2 files changed, 8 insertions(+)

diff --git a/i3lock.1 b/i3lock.1
index 761a6bd..f388a09 100644
--- a/i3lock.1
+++ b/i3lock.1
@@ -28,6 +28,7 @@ i3lock-color \- improved screen locker
 .RB [\|\-u\|]
 .RB [\|\-e\|]
 .RB [\|\-f\|]
+.RB [\|\-m\|]
 
 .SH DESCRIPTION
 .B i3lock-color
@@ -121,6 +122,11 @@ Note that this number is zero indexed. The ordering is dependent on libxinerama.
 Captures the screen and blurs it using the given sigma (radius).
 Images may still be overlaid over the blurred screenshot.
 
+.TP
+.B \-m, \-\-pass-media-keys
+Allow the following keys to be used while the screen is locked by passing them through:
+XF86AudioPlay, XF86AudioPause, XF86AudioStop, XF86AudioPrev, XF86AudioNext, XF86AudioMu
+
 .TP
 .B \-\-indicator
 Forces the indicator to always be visible, instead of only showing on activity.
diff --git a/i3lock.c b/i3lock.c
index d3fcee8..7926599 100644
--- a/i3lock.c
+++ b/i3lock.c
@@ -652,6 +652,10 @@ static void handle_key_press(xcb_key_press_event_t *event) {
             case XKB_KEY_XF86AudioMute:
             case XKB_KEY_XF86AudioLowerVolume:
             case XKB_KEY_XF86AudioRaiseVolume:
+            case XKB_KEY_XF86AudioForward:
+            case XKB_KEY_XF86AudioRewind:
+            case XKB_KEY_XF86MonBrightnessUp:
+            case XKB_KEY_XF86MonBrightnessDown:
                 xcb_send_event(conn, true, screen->root, XCB_EVENT_MASK_BUTTON_PRESS, (char *)event);
                 return;
         }
-- 
2.22.0

